postcommit.py

```diff
@@ -0,0 +1,642 @@
+
+
+COMMIT_TEMPLATE = 'LEGAL DECLARATION:I hereby ([^\n])*\n(LICENSE COMPATIBLE:([Yy][Ee][Ss]|[Nn][Oo]))*\n*'
+OPTIONAL_TEMPLATE = 'IGNORE_ISSUECHECKS:YES(\n)*(I18N ISSUE ID :)*'
+
+def getFileType(filename):
+    values = filename.split('.')
+    if len(values) > 1:
+        return values[-1]
+    else:
+        return 'others'
+
+def escapeChars(string):
+    string = string.strip('\n').strip().strip('\\').strip('/')
+    string = string.replace("\\", "\\\\").replace("'","\\'").replace("+", "\\+").replace("$", "\$")
+    return string
+
+def striplicense_template(logmsg):
+
+    description = re.sub(COMMIT_TEMPLATE,'',logmsg)
+    description = re.sub(OPTIONAL_TEMPLATE,'',description)
+    description = escapeChars(description)
+    return description
+
+def addLogs(stringVal):
+    print stringVal
+
+def getCurrentTime():
+    now = datetime.datetime.now()
+    cur_time = now.strftime('%Y-%m-%d %H:%M:%S')
+    return cur_time
+
+def status_check(exec_command, exec_dir=None, background=False):
+
+    if exec_dir:
+        os.chdir(exec_dir)
+
+    res = subprocess.Popen(exec_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if not background:
+        result = res.communicate()
+        return result
+
+class HGRepository(object):
+
+    def __init__(self, repo, ui):
+
+        self.repo_type = 'HG'
+        self.ui = ui
+        self.repo = repo
+        self.getStartChangeset()
+        self.index_node = self.repo[self.node].parents()[0].hex()[0:11]
+        self.module = os.path.basename(repo.root)
+        self.branch = self.repo['tip'].branch()
+        self.st_changeset = str(self.repo[self.node].rev()) + ":" + str(self.node)
+        self.end_changeset = str(self.repo['tip'].rev()) + ":" + str(self.repo['tip'])
+        self.commit_details = {}
+        self.branch_details = {}
+        self.diff_needed_files = {}
+        self.diff_content = {}
+
+    def getStartChangeset(self):
+        if st_changeset != "":
+            self.node = st_changeset
+        else:
+            self.node = -1
+ 
+# 	***************************************************************************************************************
+#	getCommits :
+#
+#			Initial method in this class to get commits,newchangesets json array
+#
+#			@param : 
+#			
+#			@return : {'commits': [{'files': {'Removed': [], 'Added': [{'AddedLines': '1', 'File': 'test',
+#				   'RemovedLines': '0'}], 'Modified': []}, 'changeset': '3e4983962f67', 'parents': 
+#				   'changed_lines': [1], 'file': 'test'}]}], 'user': 'gomathinayagam.d@zohocorp.com', 
+#				   'branch': 'master', 'push_time': '2018-02-28 17:43:45', 'commit_type': 1, 
+#				   'message': 'test', 'commit_date': '2018-02-28 17:43:21', 'revision': '3e49839'}], 
+#				   'newchangesets': ['3e4983962f6743f1fc55ce53e62553a5f83524c8']}
+#       						
+#	*************************************************************************************************************** 
+    def getCommits(self):
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this function to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS3776)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename method "getCommits" to match the regular expression ^[a-z_][a-z0-9_]{2,}$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS100)